### PR TITLE
Upgrade pricing library registration

### PR DESCRIPTION
**Pull request by asoong**

### CODE REVIEW

File: contracts/core/extensions/CoreInternal.sol [hunk 1]
```diff
@@ -142,7 +142,7 @@ contract CoreInternal is
      * @param  _priceLibrary   Address of contract Price Library to enable or disable
      * @param  _enabled        Whether the pricing library is enabled for use in proposal cycle
      */
-    function setPriceLibraryEnabled(
+    function registerPriceLibrary(
         address _priceLibrary,
         bool _enabled
```
Comment (felix2feng) [inline contracts/core/extensions/CoreInternal.sol]: Same comment. Why are we adding the `bool` enabled? Why do we need it?

Comment (asoong) [reply contracts/core/extensions/CoreInternal.sol]: https://github.com/SetProtocol/set-protocol-contracts/pull/254

Comment (felix2feng) [reply contracts/core/extensions/CoreInternal.sol]: Cool - as long as we have an event emitted, this works

File: contracts/core/extensions/CoreInternal.sol [hunk 2]
```diff
@@ -54,6 +54,12 @@ contract CoreInternal is
         bool _newStatus
     );
 
+    // Logs pricing library registration change. Library must conform to IAuctionPriceCurve
+    event PricingLibraryRegistrationChanged(
```
Comment (hydrazines) [inline contracts/core/extensions/CoreInternal.sol]: Nit: Change to `PriceLibraryRegistrationChanged` for consistency with function

File: contracts/core/extensions/CoreInternal.sol [hunk 3]
```diff
@@ -54,6 +54,12 @@ contract CoreInternal is
...
+        address _pricingLibrary,
```
Comment (hydrazines) [inline contracts/core/extensions/CoreInternal.sol]: Same here

**Pull request merged**